pkg/api/resource_helpers.go

```diff
@@ -97,3 +97,26 @@ func IsNodeReady(node *Node) bool {
 	}
 	return false
 }
+
+// PodRequestsAndLimits returns a dictionary of all defined resources summed up for all
+// containers of the pod.
+func PodRequestsAndLimits(pod *Pod) (reqs map[ResourceName]resource.Quantity, limits map[ResourceName]resource.Quantity, err error) {
```

Comment:
Let's try the implemented variant first. If there is resistance we can still duplicate the function, avoiding anticipatory obedience.